### PR TITLE
enhance: rewrite index params for compatibility 

### PR DESCRIPTION
**Pull request by zhagnlu**
#32900

### CODE REVIEW

File: pkg/util/indexparamcheck/index_type.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ const (
 	AutoIndex IndexType = "AUTOINDEX"
 )
 
+func IsScalarIndexType(indexType IndexType) bool {
+	return indexType == IndexSTLSORT || indexType == IndexTRIE || indexType == IndexTrie ||
+		indexType == IndexBitmap || indexType == IndexHybrid || indexType == IndexINVERTED ||
+		indexType == AutoIndex
```
Comment (czs007) [inline pkg/util/indexparamcheck/index_type.go:58]: AutoIndex can also be VectorIndex.

Comment (zhagnlu) [reply pkg/util/indexparamcheck/index_type.go:58]: OK, remove it, actually userindexparam not include autoindex

File: internal/datacoord/index_meta.go [hunk 2]
```diff
@@ -209,6 +209,7 @@ func checkParams(fieldIndex *model.Index, req *indexpb.CreateIndexRequest) bool
 		for i, param2 := range req.GetUserIndexParams() {
 			if param2.Key == param1.Key && param2.Value == param1.Value {
 				exist = true
+				break
```
Comment (czs007) [inline internal/datacoord/index_meta.go:212]: why?

File: internal/datacoord/index_meta.go [hunk 3]
```diff
@@ -224,14 +225,32 @@ func checkParams(fieldIndex *model.Index, req *indexpb.CreateIndexRequest) bool
 					}
 				}
 				exist = true
+				break
```
Comment (czs007) [inline internal/datacoord/index_meta.go:228]: why?

File: internal/datacoord/index_meta.go (hunk 2 above)
Comment (chasingegg) [reply internal/datacoord/index_meta.go:212]: have to admit this logic is not well-written, but we should not break here, need to check all the params and make sure all the params are the same

File: internal/datacoord/index_meta.go [hunk 4]
```diff
@@ -224,14 +225,32 @@ func checkParams(fieldIndex *model.Index, req *indexpb.CreateIndexRequest) bool
...
+					if param1.Key == param2.Key && param1.Value != param2.Value {
+						req.IndexParams = make([]*commonpb.KeyValuePair, len(fieldIndex.IndexParams))
+						copy(req.IndexParams, fieldIndex.IndexParams)
+						break
+					}
+				}
+			}
+		}
+	}
+	log.Debug("final request", zap.Any("create index request", req.String()))
```
Comment (chasingegg) [inline internal/datacoord/index_meta.go:253]: use info or warn log ? help to debug

File: internal/datacoord/index_meta.go (hunk 2 above)
Comment (zhagnlu) [reply internal/datacoord/index_meta.go:212]: > have to admit original logic is not well-written, but we should not break here, need to check all the params and make sure all the params are the same

acutally, outside have circle 
image
break is just break current key, outside keys always do this

File: internal/datacoord/index_meta.go (hunk 4 above)
Comment (zhagnlu) [reply internal/datacoord/index_meta.go:253]: ok

**Pull request merged**